### ISSUE TITLE
sql: TestLogic/fakedist/table: pq: internal error: job not found for table id 74, mutation 2

### ISSUE THREAD

**Issue by tbg** [cockroachdb/cockroach#36056]
Seen in CI in 19.1 on unrelated change https://github.com/cockroachdb/cockroach/pull/36026: https://teamcity.cockroachdb.com/viewLog.html?buildId=1191301&buildTypeId=Cockroach_UnitTests

```
------- Stdout: -------
=== PAUSE TestLogic/fakedist/table
            logic.go:2363: 
                
                testdata/logic_test/table:426: 
                expected success, but found
                (XX000) internal error: job not found for table id 74, mutation 2
                sql/backfill.go:470: in getJobIDForMutationWithDescriptor()
                DETAIL: stack trace:
                github.com/cockroachdb/cockroach/pkg/sql/backfill.go:470: in getJobIDForMutationWithDescriptor()
                github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1243: in markJobFailed()
                github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1262: in createRollbackJob()
                github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1193: in func2()
                github.com/cockroachdb/cockroach/pkg/sql/lease.go:421: in func1()
                github.com/cockroachdb/cockroach/pkg/internal/client/db.go:585: in func1()
                github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:688: in exec()
                github.com/cockroachdb/cockroach/pkg/internal/client/db.go:584: in Txn()
                github.com/cockroachdb/cockroach/pkg/sql/lease.go:371: in Publish()
                github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1133: in reverseMutations()
                github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:875: in rollbackSchemaChange()
                github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:860: in exec()
                github.com/cockroachdb/cockroach/pkg/sql/exec_util.go:971: in execSchemaChanges()
                github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1987: in txnStateTransitionsApplyWrapper()
                github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1306: in run()
                github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:428: in ServeConn()
                github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:338: in func4()
                
                NOTE: internal errors may have more details in logs. Use -show-logs.
            logic.go:2379: 
                testdata/logic_test/table:426: error while processing
            logic.go:2380: pq: internal error: job not found for table id 74, mutation 2
```

**Comment by knz** [comment]
I think this will fix it: #36064

**Comment by tbg** [comment]
👍 thanks
